src/sim/init_signals.hh

```diff
@@ -40,6 +42,9 @@ void initSignals();
 
 // separate out sigint handler so that we can restore the python one
 void initSigInt();
+void initSigRtmin();
+std::string extractStringFromJSON(std::string& full_str, std::string start_str,
```

Comment:
Is this really a full "json" parser or more restricted? I think it's more restricted with just {'key':'value'}.

Assuming I'm right, you should change the name to make that more clear. Also, adding some comments in the function to make the restriction clear is important as well.